examples/step-44/step-44.cc

```diff
@@ -1979,6 +1982,41 @@ namespace Step44
 
 // @sect4{Solid::get_error_dilation}
 
+// Calculate the volume of the domain in the spatial configuration
+  template <int dim>
+  void
+  Solid<dim>::update_vol_current(double &vol_current) const
```

Comment:
I think this function is misnamed, or has the wrong design. A function that's called `update_*` suggests that it updates some state. But all it updates is the argument -- it doesn't change anything else, because it's marked as `const`. My suggestion would be to call it `compute_current_volume()` and just *return* the value, rather than change the argument. After all, you don't even *use* the old value.